core/src/main/java/org/springframework/security/authentication/dao/DaoAuthenticationProvider.java

```diff
@@ -61,7 +61,11 @@ public class DaoAuthenticationProvider extends AbstractUserDetailsAuthentication
 	private UserDetailsPasswordService userDetailsPasswordService;
 
 	public DaoAuthenticationProvider() {
-		setPasswordEncoder(PasswordEncoderFactories.createDelegatingPasswordEncoder());
+		this(PasswordEncoderFactories.createDelegatingPasswordEncoder());
+	}
+
+	public DaoAuthenticationProvider(PasswordEncoder passwordEncoder) {
```

Comment:
Can you add a test where it makes sure that the provided `PasswordEncoder` is used?